src/model.ts

```diff
@@ -1151,6 +1165,45 @@ export class GitExtension implements IGitExtension {
     }
   }
 
+  /**
+   * Refresh the list of repository tags.
+   *
+   * @returns promise which resolves upon refreshing repository tags
+   */
+  async refreshTag(): Promise<void> {
+    try {
+      const data = await this._taskHandler.execute<Git.ITagResult>(
+        'git:refresh:tags',
+        async () => {
+          return await this.tags();
+        }
+      );
+
+      const tagsChanged = !JSONExt.deepEqual(
+        this._tagsList as any,
+        (data.tags ?? []) as any
+      );
+
+      this._tagsList = data.tags ?? [];
```

Comment:
Let's avoid defining the same fallback into places in case the code evolves:

```suggestion
      const newTags = data.tags ?? [];
      const tagsChanged = !JSONExt.deepEqual(
        this._tagsList as any,
        newTags as any
      );

      this._tagsList = newTags;
```